lib/opensusebasetest.pm

```diff
@@ -233,14 +233,15 @@ sub investigate_yast2_failure {
     my %y2log_errors = (
         "No textdomain configured"                   => undef,    # Detecting missing translations
                                                                   # Detecting specific errors proposed by the YaST dev team
-        "nothing provides"                           => undef,    # Detecting missing required packages
+        "(nothing provides) (?!\/bin\/sh)"           => undef,    # Detecting missing required packages
```

Comment:
interesting approach, so this matches any 'nothing provides' which is not `/bin/sh`. I guess we don't need the first capture group, right?

Reply:
i guess you need it. because there are some cases other than the ` nothing provides /bin/sh needed by product` for instance i saw things like `[zypp::solver] SATResolver.cc(problems):1207 nothing provides requested glibc`.

Reply:
in that case still matches without the parenthesis, that parenthesis just created a capture group that would not include in the full match the space after 'provides' but the full match is the same IMO.

Reply:
if i get it you are good with that?!

Reply:
not really, would you mind to explain why would not work without first parenthesis?

Reply:
i didnt say it would not work. But i dont see any problem with grouping this

Reply:
when something is not needed is better not include it at least help the user somehow, it makes easy future maintenance of that line of code. Someone could spend some time wondering if it is needed or not in order to fix it.

Reply:
PLease consider this as a better practice from my point of view.

Reply:
Could you find documentation about this? that always having capture groups on expression is a good practice, because from what I search:
"It allows to get a part of the match as a separate item in the result array." and we don't need such a thing on this case, so please better delete it or refer to some documentation where explains that having this is more correct.
If we don't find a reference, it would be code that we don't use, which is a bad practice.

Reply:
Done